packages/flutter_web_plugins/lib/src/navigation/url_strategy.dart

```diff
@@ -104,7 +104,9 @@ class HashUrlStrategy extends UrlStrategy {
     // the hash value is always prefixed with a `#`
     // and if it is empty then it will stay empty
     final String path = _platformLocation.hash;
-    assert(path.isEmpty || path.startsWith('#'));
+    // assert(path.isEmpty || path.startsWith('#'));
+
+    if (path.isNotEmpty || !path.startsWith('#')) return '/';
```

Comment:
I think you want to use an `&&` instead of `||`.

Or you could merge this with the condition below:
```dart
if (path.isEmpty || path == '#' || !path.startsWith('#')) {
  return '/';
}
```